### PR TITLE
Added support for creating a load balancer with a tag

### PR DESCRIPTION
**Pull request by shuaibmunshi**
##### SUMMARY
Adding the ability to create a load balancer with a tag. Currently load balancers can only be created with a list of droplet IDs even though the API supports tags.

##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

digital_ocean_load_balancer

##### ADDITIONAL INFORMATION

N/A

### CODE REVIEW

**Comment by shuaibmunshi** [comment]
@mamercad The sanity and unit tests should be fine now. 
I'm not quite sure what required_if is supposed to be doing but neither droplet_ids or tag are set to required because they are mutually exclusive so I removed both from required_if. If its supposed to be implemented some other way please let me know.

File: plugins/modules/digital_ocean_load_balancer.py [hunk 1]
```diff
@@ -591,6 +606,12 @@ def create(self):
                     ),
                 )
 
+        # Droplet ID and tag are mutually exclusive, check that both have not been defined
+        if self.module.params.get('droplet_ids') is not None and self.module.params.get('tag') is not None:
```
Comment (mamercad) [inline plugins/modules/digital_ocean_load_balancer.py:610]: Using [`mutally_exclusive`](https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec-dependencies) for the parameter is probably a better way to go about this.

**Comment by mamercad** [comment]
> @mamercad The sanity and unit tests should be fine now. I'm not quite sure what required_if is supposed to be doing but neither droplet_ids or tag are set to required because they are mutually exclusive so I removed both from required_if. If its supposed to be implemented some other way please let me know.

There's also a `required_one_of` which you can use ...

**Comment by mamercad** [comment]
Once you get `black / lint` passing, it'd also be great if you could add integration tests which covers both cases (covering the failing cases would be great too).

**Comment by mamercad** [comment]
Lastly -- thanks so much for working on this, we appreciate it! 🎉

File: plugins/modules/digital_ocean_load_balancer.py (hunk 1 above)
Comment (shuaibmunshi) [reply plugins/modules/digital_ocean_load_balancer.py]: Thank you for pointing this out

**Comment by shuaibmunshi** [comment]
@mamercad Sorry I forgot to test the case where tag and droplet_ids are not present and state is absent. I updated the code and ran the load_balancer integration tests myself, it should be fine now.

**Comment by shuaibmunshi** [comment]
@mamercad unless I'm missing something I don't think the failed integration tests have anything to do with my changes

**Comment by mamercad** [comment]
> @mamercad unless I'm missing something I don't think the failed integration tests have anything to do with my changes

They likely don't; I'll take a closer look when I get a bit of time -- sorry for the delay.

**Review by mamercad — CHANGES_REQUESTED** [review]
Looks good, please add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment).

**Comment by mamercad** [comment]
I checked out this branch and ran the integration tests, they passed.

**Review by mamercad — APPROVED** [review]
Thanks for this contribution!

**Pull request merged**